src/vs/workbench/contrib/bulkEdit/browser/bulkEditService.ts

```diff
@@ -148,6 +159,19 @@ class BulkEdit {
 		const model = this._instaService.createInstance(BulkCellEdits, undoRedoGroup, undoRedoSource, progress, this._token, edits);
 		return await model.apply();
 	}
+
+	private async _performCustomEdits(edits: ResourceAttachmentEdit[], undoRedoGroup: UndoRedoGroup, undoRedoSource: UndoRedoSource | undefined, progress: IProgress<void>): Promise<readonly URI[]> {
+		this._logService.debug('_performCustomEdits', JSON.stringify(edits));
+		this._undoRedoService.pushElement({
```

Comment:
the above cases create a new `Bulk__Edit` instance, which does some apply work before doing `undoRedoService.pushElement`. Not sure if this custom type has to be that involved?